src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/CodeActions/CSharp/TypeAccessibilityCodeActionProvider.cs

```diff
@@ -217,15 +217,15 @@ internal sealed class TypeAccessibilityCodeActionProvider : ICSharpCodeActionPro
 
         static bool TryGetOwner(RazorCodeActionContext context, [NotNullWhen(true)] out SyntaxNode? owner)
         {
-            var change = new SourceChange(context.Location.AbsoluteIndex, length: 0, newText: string.Empty);
             var syntaxTree = context.CodeDocument.GetSyntaxTree();
             if (syntaxTree?.Root is null)
             {
                 owner = null;
                 return false;
             }
 
-            owner = syntaxTree.Root.LocateOwner(change);
+            var token = syntaxTree.Root.FindToken(context.Location.AbsoluteIndex);
+            owner = token?.Parent;
```

Comment:
Given this is now the second PR we've needed to do `FindToken(..).Parent`, and this one needs it 3 times, I think we should create an extension method for it. `syntaxTree.Root.FindNode(..)`

Reply:
Added as a SyntaxNode extension

Reply:
![JakethedogAdventureGIF](https://github.com/dotnet/razor/assets/754264/7385fb7e-7087-4413-a461-baa37931660c)